WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java

```diff
@@ -0,0 +1,641 @@
+            AniUtils.fadeIn(cardView, AniUtils.Duration.MEDIUM);
+        } else if (newVisibility != View.VISIBLE && oldVisibility == View.VISIBLE) {
+            AniUtils.fadeOut(cardView, AniUtils.Duration.MEDIUM);
+        }
+    }
+
+    private List<?> getPlugins(@NonNull PluginListType pluginType) {
+        switch (pluginType) {
+            case POPULAR:
+                return mPluginStore.getPluginDirectory(PluginDirectoryType.POPULAR);
+            case NEW:
+                return mPluginStore.getPluginDirectory(PluginDirectoryType.NEW);
+            default:
+                return mPluginStore.getSitePlugins(mSite);
+        }
+    }
+
+    private SitePluginModel getSitePluginFromSlug(@Nullable String slug) {
+        if (slug != null) {
+            for (SitePluginModel plugin : mSitePlugins) {
+                if (slug.equals(plugin.getSlug())) {
+                    return plugin;
+                }
+            }
+        }
+        return null;
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSitePluginsFetched(PluginStore.OnSitePluginsFetched event) {
+        if (isFinishing()) {
+            return;
+        }
+        showProgress(false);
+        if (event.isError()) {
+            AppLog.e(AppLog.T.PLUGINS, "An error occurred while fetching site plugins with type: " + event.error.type);
+            ToastUtils.showToast(this, R.string.plugin_fetch_error);
+            return;
+        }
+        refreshPlugins(PluginListType.SITE);
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onWPOrgPluginFetched(PluginStore.OnWPOrgPluginFetched event) {
+        if (isFinishing()) {
+            return;
+        }
+        if (event.isError()) {
+            AppLog.e(AppLog.T.PLUGINS, "An error occurred while fetching the wporg plugin with type: " + event.error.type);
+            return;
+        }
+        if (!TextUtils.isEmpty(event.pluginSlug)) {
+            refreshPluginWithSlug(event.pluginSlug);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPluginDirectoryFetched(PluginStore.OnPluginDirectoryFetched event) {
+        if (isFinishing()) {
+            return;
+        }
+
+        if (event.isError()) {
+            AppLog.e(AppLog.T.PLUGINS, "An error occurred while fetching the plugin directory: " + event.type);
+        } else {
+            switch (event.type) {
+                case POPULAR:
+                    refreshPlugins(PluginListType.POPULAR);
+                    break;
+                case NEW:
+                    refreshPlugins(PluginListType.NEW);
+                    break;
+            }
+        }
+
+        PluginListFragment fragment = getListFragment();
+        if (fragment != null) {
+            fragment.onLoadedMore(event.canLoadMore);
```

Comment:
We need to pass the event type to the fragment, so it can ignore the value if it doesn't match it's list type. So, for example, if the popular plugins are loaded and the fragment was for the new plugins, we should ignore the `canLoadMore` value.